### PR TITLE
chore: improve package json typings

### PR DESCRIPTION
**Pull request by lifeart**
* in addition, fixed keywords field for synthetic packages
* AddonMeta properties merging

### CODE REVIEW

File: packages/compat/src/v1-app.ts [hunk 1]
```diff
@@ -436,10 +444,10 @@ export default class V1App {
         'implicit-test-styles': this.app.vendorTestStaticStyles.map(remapAsset),
         'public-assets': mapKeys(this._publicAssets, (_, key) => remapAsset(key)),
       };
-      let meta = {
+      let meta: PackageInfo = {
         name: '@embroider/synthesized-vendor',
         version: '0.0.0',
-        keywords: 'ember-addon',
+        keywords: ['ember-addon'],
```
Comment (lifeart) [inline packages/compat/src/v1-app.ts:450]: here is fix

File: packages/compat/src/v1-app.ts [hunk 2]
```diff
@@ -499,10 +507,10 @@ export default class V1App {
           addonMeta['public-assets']![`./assets/${file}`] = `/assets/${file}`;
         }
       }
-      let meta = {
+      let meta: PackageInfo = {
         name: '@embroider/synthesized-styles',
         version: '0.0.0',
-        keywords: 'ember-addon',
+        keywords: ['ember-addon'],
```
Comment (lifeart) [inline packages/compat/src/v1-app.ts:513]: here is fix

File: packages/compat/src/v1-app.ts [hunk 3]
```diff
@@ -59,7 +67,7 @@ export default class V1App {
   owningAddon: Package | undefined;
 
   static create(app: EmberApp, packageCache: PackageCache): V1App {
-    if (app.project.pkg.keywords && app.project.pkg.keywords.includes('ember-addon')) {
+    if (app.project.pkg.keywords?.includes('ember-addon')) {
```
Comment (lifeart) [inline packages/compat/src/v1-app.ts:70]: simplified condition

File: packages/shared-internals/src/metadata.ts [hunk 4]
```diff
@@ -47,3 +49,25 @@ export interface AddonMeta {
...
+  version: string;
+  main?: string;
+  peerDependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  dependencies?: Record<string, string>;
+  'ember-addon':
+    | AddonMeta
+    | AppMeta
+    | {
+        main?: string;
```
Comment (lifeart) [inline packages/shared-internals/src/metadata.ts:65]: I'm not confident about v1 version structure here

**Review by stefanpenner — APPROVED** [review]

**Pull request merged**